pkg/resourcegenerator/networking/network_policy.go

```diff
@@ -197,6 +195,22 @@ func getInboundPolicyPeers(inboundRules []podtypes.InternalRule, namespace strin
 	return policyPeers
 }
 
+func getNamespaceSelector(rule podtypes.InternalRule) *metav1.LabelSelector {
+	if rule.Namespace != "" {
+		return &metav1.LabelSelector{
+			MatchLabels: map[string]string{"kubernetes.io/metadata.name": rule.Namespace},
+		}
+	} else if rule.NamespacesByLabel != nil {
+		return &metav1.LabelSelector{
+			MatchLabels: rule.NamespacesByLabel,
+		}
+	} else {
+		return &metav1.LabelSelector{
+			MatchLabels: map[string]string{"kubernetes.io/metadata.name": rule.Namespace},
+		}
```

Comment:
this should never occur due to L#183-L#185

perhaps it would make more sense to send the application namespace to `getNamespaceSelector()` and then have the catchAll be

```go
return &metav1.LabelSelector{
	MatchLabels: map[string]string{"kubernetes.io/metadata.name": applicationNamespace},
}
```